### PR TITLE
add tree view analysis provenance details

### PR DESCRIPTION
**Pull request by myarjunar**
### What does it fix ?
* Ticket : #4074 
* Funded by : DFAT
* Description : 

selection_078

It would be great if there is any feedback @timlinux @ivanbusthomi @Charlotte-Morgan

### CODE REVIEW

**Comment by timlinux** [comment]
LGTM thanks @myarjunar 

If you can change Value Maps to Value Map that will be great.

**Comment by Charlotte-Morgan** [comment]
hi @myarjunar - this is a great addition to the analysis section of the report. I like the level of detail and the way in which you have presented the tree. 
I wonder what you think about changing the order of the elements in the tree? (this is papercut stuff)
Maybe:
- Title
- Source
- Layer purpose
- Layer Geometry
- Hazard / Exposure
- hazard category
- value map / InaSAFE fields (is InaSAFE fields the right terminology for the pop value mapping??)
- layer mode
- key word version

maybe something like InaSAFE default values should not show if there is no detail to add

**Comment by timlinux** [comment]
> maybe something like InaSAFE default values should not show if there is no detail to add

It would also be nice to see:

* Whether debug mode was enabled
* What demographic ratios were in place (i.e. keywords for hazard, exposure, aggregation layers)

File: safe/report/extractors/analysis_provenance_details.py [hunk 1]
```diff
@@ -39,51 +40,137 @@ def analysis_provenance_details_extractor(impact_report, component_metadata):
...
+        'value_maps',
+        'inasafe_fields',
+        'inasafe_default_values',
+        'layer_mode',
+        'hazard_layer',
+        'exposure_layer',
+        'aggregation_layer',
+        'keywords_version']
+
+    debug_mode = impact_report.impact_function.debug_mode
```
Comment (ismailsunni) [inline safe/report/extractors/analysis_provenance_details.py]: __[flake8]__
* [F841] local variable 'debug_mode' is assigned to but never used

File: safe/report/test/test_impact_report.py [hunk 2]
```diff
@@ -21,12 +27,15 @@
     age_displaced_count_group,
     gender_displaced_count_group)
 from safe.definitions.hazard_classifications import flood_hazard_classes
+from safe.definitions.reports import jinja2_component_type
```
Comment (ismailsunni) [inline safe/report/test/test_impact_report.py]: __[flake8]__
* [F401] 'safe.definitions.reports.jinja2_component_type' imported but unused

File: safe/report/test/test_impact_report.py [hunk 3]
```diff
@@ -21,12 +27,15 @@
...
 from safe.impact_function.impact_function import ImpactFunction
+from safe.report.processors.default import jinja2_renderer
```
Comment (ismailsunni) [inline safe/report/test/test_impact_report.py]: __[flake8]__
* [F401] 'safe.report.processors.default.jinja2_renderer' imported but unused

File: safe/report/test/hello_world_report.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+from safe.report.report_metadata import (
+    Jinja2ComponentsMetadata,
+    ReportMetadata)
+from safe.test.utilities import get_qgis_app
+from safe.utilities.i18n import tr
+
+
+# first step
+def hello_world_extractor(impact_report, component):
+    print 'component key: ' + component.key
```
Comment (ismailsunni) [inline safe/report/test/hello_world_report.py]: __[flake8]__
* [E999] SyntaxError: invalid syntax

**Comment by myarjunar** [comment]
Here is the ordered provenances

selection_081

File: safe/report/extractors/analysis_provenance_details.py (hunk 1 above)
Comment (ismailsunni) [inline safe/report/extractors/analysis_provenance_details.py]: Don't forget to fix this @myarjunar

Comment (myarjunar) [inline safe/report/extractors/analysis_provenance_details.py]: we use this to show the debug mode information

**Comment by timlinux** [comment]
LGTM thanks @myarjunar please merge

**Pull request merged**